x-pack/packages/kbn-elastic-assistant/impl/assistant/chat_send/use_chat_send.tsx

```diff
@@ -95,13 +97,17 @@ export const useChatSend = ({
       const rawResponse = await sendMessages({
         apiConfig: currentConversation.apiConfig,
         http,
-        messages: updatedMessages,
+        messages: updatedMessages ?? [],
         onNewReplacements,
         replacements: currentConversation.replacements ?? {},
       });
 
       const responseMessage: Message = getMessageFromRawResponse(rawResponse);
-      appendMessage({ conversationId: currentConversation.id, message: responseMessage });
+      await appendMessage({
+        conversationId: currentConversation.id,
+        message: responseMessage,
+      });
+      await refresh();
```

Comment:
Consider whether

```ts
await refresh()
```

should be called if

```ts
await appendMessage({
```

returns `void` (when `appendMessage` fails), because when a `console.log` is added to the following code:

```ts
      await appendMessage({
        conversationId: currentConversation.id,
        message: responseMessage,
      });
      console.log('--> calling refresh');
      await refresh();
    },
```

it appears that calling `await refresh();` causes the user's query to disappear if, for example the Kibana server is terminated via `ctrl-c`, as demonstrated in the following video:

https://github.com/elastic/kibana/assets/4459398/e2d90102-a7c3-401a-b052-f160c534ecf6

In the video above, a 2nd

```
console.log('--> calling refresh');
```

appears in the console, and the message representing the current request disappears.